### PR TITLE
Refactor 'UserWallet' types/config placement

### PR DESCRIPTION
**Pull request by vindard**
### Description

This is a cleanup PR from some of the work introduced in #321 to bring that work more in line with patterns introduced in #344. Specifically this PR:
- moves the `UserWalletConfig` into a getter function in the `config` module (similar to `SpecterWalletConfig`)

- creates a new `ITransactionLimits` interface to help remove a circular dependency between the `config.ts` and `types.ts` files. This helps to start establishing a **"`types.ts` -> `config.ts` -> `other modules`"** (DAG-like) hierarchy.

### CODE REVIEW

**Review by bodymindarts — DISMISSED** [review]

File: src/walletFactory.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { TransactionLimits, yamlConfig } from "./config"
+import { getUserWalletConfig, yamlConfig } from "./config"
```
Comment (dolcalmi) [inline src/walletFactory.ts:1]: yamlConfig use for funder is in the scope of this PR?

**Comment by krtk6160** [comment]
I'm wondering if we should allow passing config as a param, but use yamlConfig as the default value. When #321 was written, @bodymindarts mentioned that if we allow passing config as a param, we would then be able to write unit tests for them without having to change yamlConfig within the tests' scopes.

**Comment by bodymindarts** [comment]
Ah yes I forgot the unit test argument. 

But that still applies since the unit test can just create another instance of this type regardless. No need to reuse the function in this file.

Either way I think we can go with this for now for simplicity and change it later when we actually start using it in unit tests.

**Comment by bodymindarts** [comment]
Yeah I’d consider that out of scope for this

File: src/walletFactory.ts (hunk 1 above)
Comment (bodymindarts) [reply src/walletFactory.ts:1]: Yeah I’d consider that out of scope for this. Arvin is breaking the latter config refactoring into small steps.

**Review by dolcalmi — DISMISSED** [review]

**Review by bodymindarts — APPROVED** [review]

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**